### PR TITLE
Update Fruitties to Kotlin 2.0.10

### PR DESCRIPTION
**Pull request by cartland**
* Update to Kotlin 2.0.10
  * https://kotlinlang.org/docs/whatsnew20.html
* Update Room to 2.7.0-alpha06
  * https://developer.android.com/jetpack/androidx/releases/room#2.7.0-alpha06
  * https://developer.android.com/kotlin/multiplatform/room#defining-database
* Update Gradle wrapper distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-bin.zip
* Update other plugins and libraries (no code changes)

### CODE REVIEW

File: Fruitties/gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
-ksp = "1.9.23-1.0.19"
+ksp = "2.0.10-1.0.24"
 ktorVersion = "2.3.8"
-material3Android = "1.2.1"
-pagingComposeAndroid = "3.3.0-beta01"
-roomKtx = "2.6.1"
-skie = "0.6.2"
+pagingComposeAndroid = "3.3.2"
+skie = "0.8.4"
 sqlite = "2.5.0-SNAPSHOT"
```
Comment (danysantiago) [inline Fruitties/gradle/libs.versions.toml:32]: The `sqlite` version should be `2.5.0-alpha06`, I know this was not touched as part of this PR but should be cleaned up.

**Review by danysantiago — APPROVED** [review]

**Comment by cartland** [comment]
@kkuan2011 could you please take a look at this change?

File: Fruitties/shared/build.gradle.kts [hunk 2]
```diff
@@ -23,11 +26,11 @@ plugins {
 }
 
 kotlin {
-    androidTarget {
-        compilations.all {
-            kotlinOptions {
-                jvmTarget = "1.8"
-            }
+    @OptIn(ExperimentalKotlinGradlePluginApi::class)
```
Comment (kkuan2011) [inline Fruitties/shared/build.gradle.kts:29]: i presume this is the recommended way to specify the android target?

File: Fruitties/androidApp/build.gradle.kts [hunk 3]
```diff
@@ -53,6 +51,14 @@ android {
     }
 }
 
+composeCompiler {
+    enableStrongSkippingMode = true
+
+    reportsDestination = layout.buildDirectory.dir("compose_compiler")
```
[the remark this answers is not in the document]
Comment (cartland) [reply Fruitties/androidApp/build.gradle.kts:57]: This section might be unnecessary for this sample, but I got it from this page when I was trying to configure compiler options. I was actually searching for the compilerOptions that go in the shared module build file, and came across this configuration.
https://developer.android.com/develop/ui/compose/compiler#configuration-options

File: Fruitties/build.gradle.kts [hunk 4]
```diff
@@ -27,7 +28,7 @@ subprojects {
     configure<com.diffplug.gradle.spotless.SpotlessExtension> {
         kotlin {
             target("**/*.kt")
-            targetExclude("$buildDir/**/*.kt")
+            targetExclude("${layout.buildDirectory}/**/*.kt")
```
[the remark this answers is not in the document]
Comment (cartland) [reply Fruitties/build.gradle.kts:31]: Lint warned me that buildDir is deprecated and this is the replacement.

File: Fruitties/build.gradle.kts [hunk 5]
```diff
@@ -19,6 +19,7 @@ plugins {
     alias(libs.plugins.androidLibrary).apply(false)
```
Comment (javadude) [inline Fruitties/build.gradle.kts:19]: Should just use "apply false" for all of these. For example

```kotlin
alias(libs.plugins.android.library) apply false
```

File: Fruitties/gradlew [hunk 6]
```diff
@@ -1,7 +1,7 @@
-#!/usr/bin/env sh
+#!/bin/sh
```
Comment (javadude) [inline Fruitties/gradlew:1]: I assume you copied in a new gradlew?

File: Fruitties/gradle/libs.versions.toml (hunk 1 above)
Comment (cartland) [reply Fruitties/gradle/libs.versions.toml]: Done!

**Pull request merged**